controllers/toolchainstatus/toolchainstatus_controller_test.go

```diff
@@ -1297,7 +1297,230 @@ func TestToolchainStatusNotifications(t *testing.T) {
+							require.NotNil(t, notification)
+							require.Equal(t, "ToolchainStatus has now been restored to ready status for ", notification.Spec.Subject)
+							require.Equal(t, email, notification.Spec.Recipient)
+
+							t.Run("Toolchain status not ready again for extended period, notification is created", func(t *testing.T) {
+								// given
+								hostOperatorDeployment := newDeploymentWithConditions(defaultHostOperatorDeploymentName,
+									status.DeploymentNotAvailableCondition(), status.DeploymentProgressingCondition())
+
+								// Reload the toolchain status
+								require.NoError(t, fakeClient.Get(context.Background(), test.NamespacedName(test.HostOperatorNs,
+									toolchainStatus.Name), toolchainStatus))
+
+								// Reconcile in order to update the ready status to false
+								reconciler, req, fakeClient := prepareReconcile(t, requestName, newResponseGood(), mockLastGitHubAPICall, defaultGitHubClient, []string{"member-1", "member-2"},
+									hostOperatorDeployment, memberStatus, registrationServiceDeployment, toolchainStatus, toolchainConfig, proxyRouteInvalid())
+
+								// when
+								_, err := reconciler.Reconcile(context.TODO(), req)
+
+								require.NoError(t, err)
+								// Confirm there is no notification
+								assertToolchainStatusNotificationNotCreated(t, fakeClient, unreadyStatusNotification)
+								assertToolchainStatusNotificationNotCreated(t, fakeClient, restoredStatusNotification)
+
+								// Reload the toolchain status
+								require.NoError(t, fakeClient.Get(context.Background(), test.NamespacedName(test.HostOperatorNs,
+									toolchainStatus.Name), toolchainStatus))
+
+								// Now override the last transition time again
+								overrideLastTransitionTime(t, toolchainStatus, metav1.Time{Time: time.Now().Add(-time.Duration(24) * time.Hour)})
+
+								// Reconcile once more
+								reconciler, req, fakeClient = prepareReconcile(t, requestName, newResponseGood(), mockLastGitHubAPICall, defaultGitHubClient, []string{"member-1", "member-2"},
+									hostOperatorDeployment, memberStatus, registrationServiceDeployment, toolchainStatus, toolchainConfig, proxyRouteInvalid())
+
+								// when
+								res, err = reconciler.Reconcile(context.TODO(), req)
+
+								// then
+								require.NoError(t, err)
+								assert.Equal(t, requeueResult, res)
+								// Confirm restored notification is not created
+								assertToolchainStatusNotificationNotCreated(t, fakeClient, restoredStatusNotification)
+								// Confirm the unready notification has been created
+								notification := assertToolchainStatusNotificationCreated(t, fakeClient)
+								require.True(t, strings.HasPrefix(notification.ObjectMeta.Name, "toolchainstatus-unready-"))
+								require.Len(t, notification.ObjectMeta.Name, 38)
+								require.NotNil(t, notification)
+								assert.Equal(t, "ToolchainStatus has been in an unready status for an extended period for ", notification.Spec.Subject)
```

Comment:
@Kartikey-star thanks for adding the test!

How about adding just this subtest onto the existing `TestToolchainStatusNotifications` at the same level of "Toolchain status not ready again for extended period, notification is created" and maybe naming this one "Toolchain status not ready again for extended period, notification is created with invalid ProxyURL" ? In order to avoid some duplication.

Also maybe some additional common logic can be extracted to function.

Reply:
Thanks, that will make it a lot better